File: erlang/EqualWords.erl
#!/usr/bin/env escript
main(_)-> 
    {ok, Data} = file:read_file("../words.txt"),
    Anagrams = dict:new(),
    Words = binary:split(Data, [<<"\n">>], [global]),
    %lists:map(
    F = lists:foreach(
        fun(Word)-> 
            WordString = binary_to_list(Word),
            io:format(WordString),
            io:format("-"),
            WordSorted = lists:sort(WordString),

            io:format(WordSorted),
            io:format("\n"),
            WordSorted
            %Anagrams = lists:append(WordSorted, WordString, Anagrams)
            %dict:append("WordSorted", "Wordstring", Anagrams)
        end, 
        Words),
    %WordsSeperated = lists:join(",", Sorted),
    io:format("|||||||\n"),
    io:format(dict:fetch_keys(Anagrams)).

File: php/equalwords.php
<?php
$file = $argv[1];
$words = file($file, FILE_IGNORE_NEW_LINES);

$anagrams = [];

foreach ($words as $word) {
    $word_split = str_split($word);
    sort($word_split);
    $word_sorted = implode($word_split);
    
    $anagrams[$word_sorted][] = $word;
}

foreach ($anagrams as $anagram) {
    echo implode(', ', $anagram) . "\n";
}
